core/frontend/src/IModelConnection.ts

```diff
@@ -907,6 +922,62 @@ export namespace IModelConnection { // eslint-disable-line no-redeclare
       const iModel = this._iModel;
       return iModel.isOpen ? IModelReadRpcInterface.getClientForRouting(iModel.routingContext.token).queryElementProps(iModel.getRpcProps(), params) : [];
     }
+
+    public async getPlacements(elementIds: Iterable<Id64String>): Promise<Array<Placement & { elementId: Id64String }>> {
+      let ids: Id64String[];
+      if (typeof elementIds === "string")
+        ids = [elementIds];
+      else if (!Array.isArray(elementIds))
+        ids = Array.from(elementIds);
+      else
+        ids = elementIds;
+
+      if (ids.length === 0)
+        return [];
+
+      const ecsql = `
+        SELECT
+          ECInstanceId,
+          Origin.x as x, Origin.y as y, Origin.z as z,
+          BBoxLow.x as lx, BBoxLow.y as ly, BBoxLow.z as lz,
+          BBoxHigh.x as hx, BBoxHigh.y as hy, BBoxHigh.z as hz,
+          Yaw, Pitch, Roll,
+          NULL as Rotation
+        FROM bis.GeometricElement3d WHERE Origin IS NOT NULL AND BBoxLow IS NOT NULL AND BBoxHigh IS NOT NULL
+        UNION ALL
+        SELECT * FROM (
+          SELECT
+            ECInstanceId,
+            Origin.x, Origin.y, NULL,
+            BBoxLow.x, BBoxLow.y, NULL,
+            BBoxHigh.x, BBoxHigh.y, NULL,
+            NULL, NULL, NULL,
+            Rotation
+          FROM bis.GeometricElement2d WHERE Origin IS NOT NULL AND BBoxLow IS NOT NULL AND BBoxHigh IS NOT NULL)
+        WHERE ECInstanceId IN (${ids.join(",")})

```

Comment:
probably best to make ids a parameter and bind the string (avoids SQL injection risk).

Reply:
`WHERE ECInstanceId IN ?` => BE_SQLITE_ERROR: InvalidECSql: not an error (BE_SQLITE_OK)
`WHERE ECInstanceId IN (?)` => BE_SQLITE_ERROR: Failed to bind parameter '1'
with ids passed as only binding.
`for await (let row of this._iModel.query(ecsql, [ids]))`
I can't test bindings in iModelConsole to try and get an actual useful error message...

Reply:
Does anyone other than @khanaffan know if bindings are supported for `IN`?

Reply:
@ColinKerr?

Reply:
did you try `ids.join(",")` as the binding?

Reply:
Yes. That is successfully parsed, but produces no results, presumably because it compares the ECInstanceId to my one big long string.

Reply:
There is no way to bind these Ids as a parameter from the frontend. That sounds like an important and useful future enhancement.

Reply:
should probably also allow `compressedIdSet`

Reply:
That's just an alias for string. API exists to obtain Iterable<Id64String> from it.